io/io/src/TFileMergeClient.cxx

```diff
@@ -0,0 +1,78 @@
+#include "TArrayC.h"
+#include "TBufferFile.h"
+#include "TFileMergeClient.h"
+#include "TFileMergeServer.h"
+
+namespace ROOT {
+namespace Experimental {
+
+TFileMergeClient::TFileMergeClient(TFileMergeServer& s)
+   : TMemFile(s.GetOutputName().c_str(), "recreate"), fServer(s), fClassSent(0) {}
+
+TFileMergeClient::~TFileMergeClient()
+{
+   TMemFile::Close();
+   delete fClassSent;
+}
+
+Int_t TFileMergeClient::Write(const char *name, Int_t opt, Int_t bufsize)
+{
+   Int_t nbytes = TMemFile::Write(name, opt, bufsize);
+
+   if (nbytes) {
+      TBufferFile *fBuffer = new TBufferFile(TBuffer::kWrite);
+
+      fBuffer->WriteTString(GetName());
+      fBuffer->WriteLong64(GetEND());
+      CopyTo(*fBuffer);
+
+      fServer.Push(fBuffer);
+
+      // Record StreamerInfo sent to the server
+      Int_t isize = fClassIndex->GetSize();
+      if (!fClassSent) {
+         fClassSent = new TArrayC(isize);
+      } else {
+         if (isize > fClassSent->GetSize()) {
+            fClassSent->Set(isize);
+         }
+      }
+      for (Int_t c = 0; c < isize; ++c) {
+         if (fClassIndex->fArray[c]) {
+            fClassSent->fArray[c] = 1;
+         }
+      }
+      ResetAfterMerge(0);
+   }
+   return nbytes;
+}
+
+Int_t TFileMergeClient::Write(const char *, Int_t, Int_t) const
+{
+   Error("Write const TFile", "A const TFile object cannot not be saved.");
+   return 0;
+}
+
+void TFileMergeClient::WriteStreamerInfo()
+{
+   if (!fWritable) return;
```

Comment:
If it was not for the fact that WriteStreamerInfo is a public routine, this first statement may as well be an assert (all calls to WriteStreamerInfo are either behind a if (fWritable) or a if (!fWriteable) { Error(...); return; } )